constants/src/lib.rs

```diff
@@ -6,6 +6,7 @@ pub const DBMS_NAME: &str = "MongoDB Atlas";
 pub const NOT_IMPLEMENTED: &str = "HYC00";
 pub const TIMEOUT_EXPIRED: &str = "HYT00";
 pub const GENERAL_ERROR: &str = "HY000";
+pub const PROGRAM_TYPE_OUT_OF_RANGE: &str = "HY003";
```

Comment:
This error is listed as (DM) [here](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlgetdata-function?view=sql-server-ver16) so it might be wrong for me to include it in our code and return it when the `SQLGetData` `target_type` is out of range. Not sure, so please let me know!